sns/parity_math.c

```diff
@@ -1478,38 +1563,42 @@ static void fail_idx_xor_recover(struct m0_parity_math *math,
                         ((uint8_t*)parity[0].b_addr)[ei] = pe;
         }
 
+	return M0_RC(0);
 }
 
 #if !ISAL_ENCODE_ENABLED
 /** @todo Iterative reed-solomon decode to be implemented. */
-static void fail_idx_reed_solomon_recover(struct m0_parity_math *math,
-					  struct m0_buf *data,
-					  struct m0_buf *parity,
-					  const uint32_t failure_index)
+static int fail_idx_reed_solomon_recover(struct m0_parity_math *math,
+					 struct m0_buf *data,
+					 struct m0_buf *parity,
+					 const uint32_t failure_index)
 {
+	return M0_RC(0);
 }
 #endif /* !ISAL_ENCODE_ENABLED */
 
 #if ISAL_ENCODE_ENABLED
-static void fail_idx_isal_recover(struct m0_parity_math *math,
-				  struct m0_buf *data,
-				  struct m0_buf *parity,
-				  const uint32_t failure_index)
+static int fail_idx_isal_recover(struct m0_parity_math *math,
```

Comment:
Since you are not returning anything from this function you do not want to have a return value as int.

Reply:
It is required to have return value as 'int' here because it is getting called inside `m0_parity_math_fail_index_recover()` using array of function pointers. Here, we are not returning anything because we are asserting here forcefully saying 'This functionality is not implemented'.